interfaces/inc/c_pal/execution_engine.h

```diff
@@ -3,12 +3,27 @@
 #ifndef EXECUTION_ENGINE_H
 #define EXECUTION_ENGINE_H
 
+#ifdef __cplusplus
+#include <cstdint>
+#else
+#include <stdint.h>
+#endif
 
 #include "umock_c/umock_c_prod.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef struct EXECUTION_ENGINE_PARAMETERS_TAG
+{
+    uint32_t min_thread_count;
+    uint32_t max_thread_count;
+} EXECUTION_ENGINE_PARAMETERS;
+
+#define DEFAULT_MIN_THREAD_COUNT 4
+#define DEFAULT_MAX_THREAD_COUNT 0 // no max thread count
+
 typedef struct EXECUTION_ENGINE_TAG* EXECUTION_ENGINE_HANDLE;
 
 MOCKABLE_FUNCTION(, EXECUTION_ENGINE_HANDLE, execution_engine_create, void*, execution_engine_parameters);
```

Comment:
>void [](http://example.com/codeflow?start=70&length=4)

isn't this then const `EXECUTION_ENGINE_PARAMETERS`*?

Reply:
I believe it can be.